src/Symfony/Component/Translation/Bridge/Phrase/Config/WriteConfig.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\Phrase\Config;
+
+use Symfony\Component\Translation\Provider\Dsn;
+
+/**
+ * @psalm-type PhraseWriteConfig array{
+ *  autotranslate?: string,
+ *  file_encoding?: string,
+ *  file_format: string,
+ *  format_options?: array<string, string>,
+ *  locale_id: string,
+ *  locale_mapping?: array<string, string>,
+ *  mark_reviewed?: string,
+ *  skip_unverification?: string,
+ *  skip_upload_tags?: string,
+ *  tags: string,
+ *  update_descriptions?: string,
+ *  update_translations: string,
+ * }
+ * @psalm-type PhraseDsnWriteConfig array{
+ *  autotranslate?: string,
+ *  file?: string,
+ *  file_encoding?: string,
+ *  file_format?: string,
+ *  format_options?: array<string, string>,
+ *  locale_id?: string,
+ *  locale_mapping?: array<string, string>,
+ *  mark_reviewed?: string,
+ *  skip_unverification?: string,
+ *  skip_upload_tags?: string,
+ *  tags?: string,
+ *  update_descriptions?: string,
+ *  update_translations?: string,
+ * }
+ *
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+class WriteConfig
+{
+    private const DEFAULTS = [
+        'file_format' => 'symfony_xliff',
+        'update_translations' => '1',
+    ];
+
+    /**
+     * @psalm-param PhraseWriteConfig $options
+     */
+    private function __construct(
+        private array $options,
+    ) {
+    }
+
+    public function withTag(string $tag): self
+    {
+        $this->options['tags'] = $tag;
+
+        return $this;
+    }
+
+    public function withLocale(string $locale): self
```

Comment:
set* (same for all methods in the PR :) )
and all should return `static` instead of `self` + have the `@return $this` annotation BTW!